doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+inference data to create masks, then applies filters and composes the output
+retouched image. What is each of these parts for? To be more certain, the
+"beautification" implemented here is smoothing out the skin (by the
+`Bilateral filter`) and making eyes, a nose and a mouth sharper (by the
+`Unsharp mask`). Thus, to know which filter should be applied to which area of
+image, we need to calculate a contour of the face and contours of facial
+elements --- we can get them from data received via deep learning methods.
+
+Two topologies from OMZ have been used in this sample: the
+<a href="https://github.com/opencv/open_model_zoo/tree/master/models/intel
+/face-detection-adas-0001">face-detection-adas-0001</a>
+and the
+<a href="https://github.com/opencv/open_model_zoo/blob/master/models/intel
+/facial-landmarks-35-adas-0002/description/facial-landmarks-35-adas-0002.md">
+facial-landmarks-35-adas-0002</a>.
+
+The face detector takes the input image and returns a blob with the shape
+[1, 1, N, 7] after the inference, where `N = 200` is the maximum number of
+faces which can be detected. Obviously, the output `cv::Mat` post-processing is
+required. The structure of the output for every face is the following:
+`[image_id, label, conf, x_min, y_min, x_max, y_max]`,
+all the seven elements are floating point. For our further goals we need to take
+only results with `image_id > 0` because a negative value in this field
+indicates the end of detections; also we can cut detections by the `conf` field
+to avoid mistakes of the detector. The last four float numbers we've got denote
+normalized coordinates and are between 0 and 1; to get the real pixel
+coordinates we should multiply it by the image sizes respectively to the
+directions.
+
+Then, to create masks we need to get landmarks of facial elements for every
+detected face. The chosen detector expects to obtain square area with only one
+face contained so we have to infer the model separately for every ROI we've
+got from the previous detection. The output of the facial landmarks detector is
+a blob with 35 landmarks: the first 18 of them are facial elements
+(eyes, eyebrows, a nose, a mouth) and the last 17 --- a jaw contour. Landmarks
+are floating point values of coordinates normalized relatively to an input ROI
+(not the original frame). In addition, for the further goals we need contours of
+eyes, mouths, faces, etc., not the landmarks. So, post-processing of the Mat is
+also required here. The process is split into two parts --- landmarks'
+coordinates denormalization to the real pixel coordinates of the source frame
+and getting necessary closed contours based on these coordinates.
+
+The last step of processing the inference data is drawing masks using the
+received contours. Honestly speaking, there is no need to get the most accurate
+contours in the world: we can always blur masks borders by gaussian filter to
+occupy a bit more area. Another point that should be mentioned here is getting
```

Comment:
"In this demo the contours don't need to be pixel accurate, since masks are blurred with Gaussian filter anyway."